File: membershipDB/src/main/java/Member.java
public record Member(int id, Grade grade, String name, String email, String phone) {
    @Override
    public boolean equals(Object o) {
        if(this == o) return true;
        if(!(o instanceof Member member)) return false;
        return this.email.equals(member.email);
    }
    @Override
    public int hashCode() {
        return email.hashCode();
    }

    @Override
    public String toString() {
        return id + "/ " + grade + "/ " + name + "/ " + email + "/ " + phone;
    }
}

File: kotlin/kotlin-part-3/src/main/kotlin/member/Member.kt
package member

data class Member(val name: String, val email: String, val phone: String) {

    // 여기서 toString을 override 해 버리면 자동 생성된 것이 사라진다.
    // 디버깅용 출력은 남겨 두는 편이 낫기 때문에, 화면용은 따로 프로퍼티로 만든다.
    val display: String
        get() = "[이름] $name, [이메일] $email, [연락처] $phone"

}